bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -149,6 +150,8 @@
     private volatile boolean running = false;
     // Flag identify whether it is in shutting down progress
     private volatile boolean shuttingdown = false;
+    // Bookie status
+    private BookieStatus bookieStatus = new BookieStatus();
```

Comment:
@jvrao : this allows turn a bookie to readonly or writable on-the-fly and keep the bookie status regarding restarts. so you can hook this with http endpoint and your deployment/administration tools to do automatic operations.